core/src/test/java/com/datastrato/gravitino/listener/api/event/TestPartitionEvent.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datastrato.gravitino.listener.api.event;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.catalog.PartitionDispatcher;
+import com.datastrato.gravitino.exceptions.GravitinoRuntimeException;
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.listener.DummyEventListener;
+import com.datastrato.gravitino.listener.EventBus;
+import com.datastrato.gravitino.listener.PartitionEventDispatcher;
+import com.datastrato.gravitino.listener.api.info.PartitionInfo;
+import com.datastrato.gravitino.rel.partitions.Partition;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public class TestPartitionEvent {
+  private PartitionDispatcher dispatcher;
+  private PartitionDispatcher failureDispatcher;
+  private DummyEventListener dummyEventListener;
+  private Partition partition;
+
+  @BeforeAll
+  void init() {
+    this.partition = mockPartition();
+    this.dummyEventListener = new DummyEventListener();
+    EventBus eventBus = new EventBus(Arrays.asList(dummyEventListener));
+    PartitionDispatcher partitionDispatcher = mockPartitionDispatcher();
+    this.dispatcher = new PartitionEventDispatcher(eventBus, partitionDispatcher);
+    PartitionDispatcher partitionExceptionDispatcher = mockExceptionPartitionDispatcher();
+    this.failureDispatcher = new PartitionEventDispatcher(eventBus, partitionExceptionDispatcher);
+  }
+
+  @Test
+  void testAddPartitionEvent() {
+    NameIdentifier identifier = NameIdentifier.of("metalake", partition.name());
```

Comment:
should be `NameIdentifier.of("metalake","catalog","schema","table")`, not include `partition.name()`